src/main/java/ortus/boxlang/runtime/bifs/global/list/ListRest.java

```diff
@@ -58,22 +59,40 @@ public ListRest() {
 	 *
 	 * @argument.delimiter string the list delimiter
 	 *
-	 * @argument.includeEmptyFields boolean whether to include empty fields in the returned result
+	 * @argument.includeEmptyFields boolean whether to include empty fields in the
+	 *                              returned result
 	 *
-	 * @argument.multiCharacterDelimiter boolean whether the delimiter is multi-character
+	 * @argument.multiCharacterDelimiter boolean whether the delimiter is
+	 *                                   multi-character
 	 */
-	public Object _invoke( IBoxContext context, ArgumentsScope arguments ) {
-		Integer	offset	= arguments.getAsInteger( Key.offset );
-		Array	ref		= ListUtil.asList(
-		    arguments.getAsString( Key.list ),
-		    arguments.getAsString( Key.delimiter ),
-		    arguments.getAsBoolean( Key.includeEmptyFields ),
-		    true
-		);
-		if ( ref.size() >= 1 ) {
-			ref.remove( 0 + offset );
+	public Object _invoke(IBoxContext context, ArgumentsScope arguments) {
+		String list = arguments.getAsString(Key.list);
+		String delimiter = arguments.getAsString(Key.delimiter);
+		Boolean includeEmptyFields = arguments.getAsBoolean(Key.includeEmptyFields);
+		Boolean multiCharacterDelimiter = arguments.getAsBoolean(Key.multiCharacterDelimiter);
+		Integer offset = arguments.getAsInteger(Key.offset);
+
+		Array ref = ListUtil.asList(list, delimiter, includeEmptyFields, multiCharacterDelimiter);
+		if (ref.size() >= 1) {
```

Comment:
This has the potential of an index out of bounds now that I think of it, if the offest is > than the ref.size().

I would suggest:

```
if (offset >= 0 && offset < ref.size()) {
	ref.remove(offset);
}
```

Reply:
I'd hold off on this-- I'm recommending we completely remove this logic anyway in favor of a totally different approach.  See my comments.